src/foam/util/NUIDGenerator.js

```diff
@@ -19,26 +19,51 @@ foam.CLASS({
     'foam.core.PropertyInfo',
     'foam.dao.AbstractSink',
     'foam.dao.DAO',
+    'foam.mlang.order.Comparator',
     'static foam.util.UIDSupport.*'
   ],
 
   javaCode: `
-  public NUIDGenerator(X x, String salt) {
+  public NUIDGenerator(X x, String salt, DAO dao, PropertyInfo pInfo) {
     setX(x);
     setSalt(salt);
+    setDao(dao);
+    setPropertyInfo(pInfo);
+    init_();
+  }
+
+  public NUIDGenerator(X x, String salt, DAO dao, PropertyInfo pInfo, Comparator comparator) {
+    setX(x);
+    setSalt(salt);
+    setDao(dao);
```

Comment:
might setDao before setSalt then the postset doesn't have to lookup the dao just to be replaced.

Reply:
yes, but no control over Builder, so doesn't really matter, and it's only one time.